runtime/onert/core/src/ir/Shape.cc

```diff
@@ -73,9 +73,10 @@ void Shape::extendRank(int to_rank)
 
 uint64_t Shape::num_elements() const
 {
-  // All of the nodes must have non-negative dimension
-  assert(std::all_of(_dimensions.begin(), _dimensions.end(),
-                     [](const int32_t &v) { return (v >= 0); }));
+  // if dimension is 0, it means unspecified and cannot calculate the total number of elements
+  if (std::any_of(_dimensions.begin(), _dimensions.end(),
+                  [](const int32_t &v) { return (v == 0); }))
+    throw std::runtime_error("num_elements() cannot calculate when a dimension is unspecified");
```

Comment:
If the operand is a scalar, is there no problem?

Reply:
To make that clear, I just added a test case :-)
- https://github.com/Samsung/ONE/pull/1547/files#diff-4280003f96f8b6faf02d7708e456cf48R36

Reply:
Ah, I was mistaken for it. Thanks.